src/Adapter/Shipment/CommandHandler/MergeProductsToShipmentHandler.php

```diff
@@ -59,10 +60,18 @@ public function handle(MergeProductsToShipment $command)
             throw new ShipmentNotFoundException(sprintf('Shipment with id "%s" was not found', $sourceId));
         }
 
+        if (!empty($sourceShipment->getTrackingNumber())) {
+            throw new CannotEditShipmentShippedException(sprintf('Cannot merge shipment "%s" it has already been shipped.', $sourceId));
+        }
+
         if (!$targetShipment) {
             throw new ShipmentNotFoundException(sprintf('Shipment with id "%s" was not found', $targetId));
         }
 
+        if (!empty($targetShipment->getTrackingNumber())) {
+            throw new CannotEditShipmentShippedException(sprintf('Cannot merge into shipment "%s" it has already been shipped.', $targetId));
```

Comment:
```suggestion
            throw new CannotEditShipmentShippedException(sprintf('Cannot merge into shipment "%s" because it has already been shipped.', $targetId));
```